### PR TITLE
media-playback: Add speed percentage

### PR DESCRIPTION
**Pull request by exeldro**
Allow adjusting the playback speed of media source.

### CODE REVIEW

**Comment by jp9000** [comment]
Why?

**Comment by exeldro** [comment]
I use it for slow motion replays

File: deps/media-playback/media-playback/decode.c [hunk 1]
```diff
@@ -336,6 +336,15 @@ bool mp_decode_next(struct mp_decode *d)
 					d->stream->time_base,
 					(AVRational){1, 1000000000});
 
+		
+		d->frame_pts = av_rescale_q(d->frame_pts,
+			(AVRational) {1, d->m->speed_percentage},
+			(AVRational) {1, 100});
```
Comment (RytoEX) [inline deps/media-playback/media-playback/decode.c]: Style:  Indent line continuations with two tabs (Lines 341-342).

Don't put a space between `(AVRational)` and the opening curly brace after it.

File: plugins/obs-ffmpeg/obs-ffmpeg-source.c [hunk 2]
```diff
@@ -186,6 +188,9 @@ static obs_properties_t *ffmpeg_source_getproperties(void *data)
 
 	obs_properties_add_bool(props, "seekable", obs_module_text("Seekable"));
 
+	obs_properties_add_int_slider(props, "speed_percentage",
+		obs_module_text("SpeedPercentage"), 1, 10000, 1);
```
Comment (RytoEX) [inline plugins/obs-ffmpeg/obs-ffmpeg-source.c]: Style: Indent line continuations with two tabs.

File: deps/media-playback/media-playback/decode.c [hunk 3]
```diff
@@ -336,6 +336,15 @@ bool mp_decode_next(struct mp_decode *d)
...
+
+		duration = av_rescale_q(duration,
+				(AVRational) {1, d->m->speed_percentage},
+				(AVRational) {1, 100});
```
Comment (RytoEX) [inline deps/media-playback/media-playback/decode.c]: Style:  Don't put a space between (AVRational) and the opening curly brace after it.

**Comment by jp9000** [comment]
I got rid of the large parameter list for the mp_media_init however.  Merging shortly, you'll see "closed in [commit]".

**Comment by paulpv** [comment]
@exeldro How to use this feature?
I see in the code that it adds a `obs_properties_add_int_slider(props, "speed_percentage" ...`, but I am using OBS 27.2.4 (mac) and I do not see any "Speed" option in any `Media Source` properties.
However, I also don't see the existing `Looping` option, which I also want, so I must be missing something obvious.
image

**Pull request closed without merging**